StoryboardsAndXibs.md

```diff
@@ -10,18 +10,42 @@ In most cases on both macOS and iOS, it is recommended to avoid storyboards (and
     - [Compose view hierarchies With NSStackViews/UIStackViews](Layout.md#Compose-Hierarchies-with-NSStackViews/UIStackViews).
     - [Use layout anchor constraints](Layout.md#Use-Layout-Anchor-Constraints)
     - [Create the view hierarchy in loadView](ViewControllers.md#Construct-View-Hierarchies-in-loadView)
+- Not supported in VisionOS
 
 ## Exceptions
-- Launch Screens on iOS are required to be specified as storyboards.
 - The main menu bar for a macOS app is required to be specified in a XIB.
 
-## How to convert xib to programmatic initialization
+## How to convert xib/storyboards to programmatic initialization on iOS
+- Remove any reference to loading UIStoryboard pointer [storyboardWithName:bundle:](https://developer.apple.com/documentation/uikit/uistoryboard/1616216-storyboardwithname)
 - Change the UIViewController caller
-    - Instead of passing in [nibName](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621359-init) in particular bundle, pass in nil
-    - Make sure no other storyboard is initializing with [iniWithCoder:](https://developer.apple.com/documentation/foundation/nscoding/1416145-initwithcoder?language=objc)
+    - If the UIViewController was initialized by storyboard, instead of [instantiateViewController(withIdentifier:)](https://developer.apple.com/documentation/uikit/uistoryboard/1616214-instantiateviewcontroller), use [init(nibName:bundle:)](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621359-init) with parameter nil for nibName and bundle.
+    - If the UIViewController was initialized by xibs, instead of passing in [init(nibName:bundle:)](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621359-init) in particular bundle, pass in nil for nibName and bundle.
+    - Make sure no content of storyboard is initializing with [initWithCoder:](https://developer.apple.com/documentation/foundation/nscoding/1416145-initwithcoder?language=objc)
+- For storyboards, make sure segues related functions do NOT get called. 
+    - Do NOT Use
+        - [shouldPerformSegue(withIdentifier:sender:)](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621502-shouldperformsegue)
+        - [prepare(for:sender:)](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621490-prepare)
+        - [allowedChildrenForUnwinding(from:)](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621371-allowedchildrenforunwinding)
+        - [childContaining(_:)](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621384-childcontaining)
+        - [canPerformUnwindSegueAction(_:from:sender:)](https://developer.apple.com/documentation/uikit/uiviewcontroller/3089101-canperformunwindsegueaction)
+        - [unwind(for:towards:)](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621473-unwind)
+    - Can use
```

Comment:
Should we add viewIsAppearing to this list?

Reply:
Though maybe we don't want to be in the business of adding every view lifecycle method to this list. Maybe it's more about explaining which ones not to use or saying, here are some common 1:1ish API replacements?

Reply:
hmmm its hard to determine the 1;1 easily without knowing all the logic behind it....